composer/datasets/utils.py

```diff
@@ -166,3 +170,63 @@ def add_vision_dataset_transform(dataset: VisionDataset, transform: Callable, is
         else:
             dataset.transform = transforms.Compose([dataset.transform, transform])
             log.warning(transform_added_logstring)
+
+
+try:
```

Comment:
@dakinggg  what is the proper way to handle subclassing a class from `transformers`? Is it possible to do a conditional import, or does code like this make transformers a required dependency?

Reply:
@abhi-mosaic  do you know how to handle this?

Reply:
@mosaicml/composer-team-eng any thoughts on how to handle subclassing from a class defined in a conditional import?

Reply:
I think you'll need to conditionally define the superclass. that is, in the except case, set the class to `None`, and then error properly on construction if the dependency is missing.